File: docs/PayAPP-연동API-oapi-0043/examples/SAMPLE_ANDROID/SAMPLE_ANDROID/payapp_android_sample_v3/OApi/src/main/java/com/udid/oapi/data/PASubidRegistData.java
package com.udid.oapi.data;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import lombok.ToString;

/**
 * Create by 김진원
 * 판매자 회원 가입
 */
@Getter
@Setter
@ToString
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class PASubidRegistData {

    /**
     * 판매자 회원 아이디
     * 필수
     */
    @NonNull
    String userid;

    /**
     * 부계정 아이디
     * 필수
     */
    @NonNull
    String subuserid;

    /**
     * 부계정 비밀번호
     * 필수
     */
    @NonNull
    String subpwd;

    /**
     * 부계정명
     * 필수
     */
    @NonNull
    String subname;

    /**
     * 대분류
     */
    String subetc1;

    /**
     * 중분류
     */
    String subetc2;

    /**
     * 소분류
     */
    String subetc3;

    /**
     * 직급
     */
    String subjtype;

    /**
     * 권한
     */
    String state;


    private static PASubidRegistDataBuilder builder() {
        return new PASubidRegistDataBuilder();
    }

    /**
     * 필수 값
     * @param userid
     * @param subuserid
     * @param subpwd
     * @param subname
     * @return
     */
    public static PASubidRegistDataBuilder builder(String userid, String subuserid, String subpwd, String subname) {

        return PASubidRegistData.builder().userid(userid).subuserid(subuserid).subpwd(subpwd).subname(subname);
    }
}

File: docs/PayAPP-연동API-oapi-0043/examples/SAMPLE_IOS/SAMPLE_IOS/payapp_ios_sample_v3/OAPI/model/data/PASubidRegistData.swift
//
//  PASubidRegistData.swift
//  Payapp_iOS_Sample_V2
//
//  Created by Jin Won Kim on 2021/11/01.
//

import ObjectMapper

class PASubidRegistData: Mappable {

    /**
    * cmd
    * 필수
    */
    var cmd:String?
    
    /**
     * 판매자 회원 아이디
     * 필수
     */
    var userid:String?

    /**
     * 부계정 아이디
     * 필수
     */
    var subuserid:String?

    /**
     * 부계정 비밀번호
     * 필수
     */
    var subpwd:String?

    /**
     * 부계정명
     * 필수
     */
    var subname:String?

    /**
     * 대분류
     */
    var subetc1:String?

    /**
     * 중분류
     */
    var subetc2:String?

    /**
     * 소분류
     */
    var subetc3:String?

    /**
     * 직급
     */
    var subjtype:String?

    /**
     * 권한
     */
    var state:String?

    required init?(map: Map){}
    init(userid:String, subuserid:String, subpwd:String, subname:String){
        
        self.userid = userid
        self.subuserid = subuserid
        self.subpwd = subpwd
        self.subname = subname
    }
    
    func mapping(map: Map) {
        
        cmd                             <- map["cmd"]
        userid                          <- map["userid"]
        subuserid                       <- map["subuserid"]
        subpwd                          <- map["subpwd"]
        subname                         <- map["subname"]
        subetc1                         <- map["subetc1"]
        subetc2                         <- map["subetc2"]
        subetc3                         <- map["subetc3"]
        subjtype                        <- map["subjtype"]
        state                           <- map["state"]

    }
}
